### DISCUSSION TITLE
Executing cli commands programmatically

### DISCUSSION THREAD

**Discussion by willruggiano** [prisma/prisma#18996] [Q&A] [answered]
I'd like to create a Lambda function that can run the equivalent of `prisma migrate deploy`, but programmatically (from NodeJS) instead of as a shell command.

The usecase is this:
I have a fairly complex CI pipeline that I am deploying in AWS. When I say "fairly complex", I mean that there are multiple cross-account stacks being deployed by a single pipeline.
As part of that deployment process, I'd like to `prisma migrate deploy` to each stack's RDS cluster. These clusters exist in a private subnet (and I don't want them to be public, for obvious reasons) so I need to execute `prisma migrate deploy` from within the cluster's VPC. I've come up with several potential solutions and the "least hacky" seems to be deploying a Lambda function with the Prisma client/sdk bundled and programmatically invoking the equivalent of `prisma migrate deploy`. Anyways, I saw the warning ("This package is intended for Prisma's internal use") in the README for prisma/packages/migrate and thought I'd ask.

TL;DR I'd like to be able to run `prisma migrate deploy` using Prisma's TypeScript sdk, rather than as a shell command.

**Comment by nurul3101** [comment, marked as answer]
Hello @willruggiano 👋

Can you have a look at this GitHub Issue

- https://github.com/prisma/prisma/issues/4703

There are a few suggestions on how you can execute migrate commands programmatically.

**Reply by willruggiano** [reply]
Ah yes that is what I am looking for. Or at least on topic. Thanks!

**Reply by nurul3101** [reply]
No worries! 🙂
